src/subapps/dataExplorer/PredicateSelector.tsx

```diff
@@ -142,11 +176,47 @@ const getPathsForResource = (
   return paths;
 };
 
+export const checkPathExistence = (
+  resource: { [key: string]: any },
+  path: string,
+  criteria: 'exists' | 'does-not-exist' = 'exists'
+): boolean => {
+  if (path in resource) {
+    return criteria === 'exists' ? true : false;
+  }
+
+  const subpaths = path.split('.');
+
+  for (const subpath of subpaths) {
+    const valueAtSubpath = resource[subpath];
+    const remainingPath = subpaths.slice(1);
+    if (!(subpath in resource)) {
+      return criteria === 'exists' ? false : true;
+    }
+
+    if (Array.isArray(valueAtSubpath)) {
+      return valueAtSubpath.some(value =>
+        checkPathExistence(value, remainingPath.join('.'), criteria)
+      );
+    }
+    if (isObject(valueAtSubpath)) {
+      return checkPathExistence(
+        valueAtSubpath,
+        remainingPath.join('.'),
+        criteria
+      );
+    }
+    break;
+  }
+
+  return criteria === 'exists' ? false : true;
+};
+
 export const isPathMissing = (
   resource: { [key: string]: any },
   path: string
 ): boolean => {
-  if (path in resource) {
+  if (path in resource && path !== '') {
```

Comment:
Is there situation where the path is empty string ?
If so is it better to filter paths when you generate them (just a suggestion)